File: tests/positives/framework-detection/kotlin/detect-gemini.kt
// Positive test cases for Gemini (Kotlin)

// ruleid: detect-gemini
import com.google.ai.client.generativeai.GenerativeModel

// ruleid: detect-gemini
val model = GenerativeModel(modelName = "gemini-pro", apiKey = "key")

// ruleid: detect-gemini
val response = model.generateContent("Hello")

// ruleid: detect-gemini
val chat = model.startChat()

File: tests/positives/framework-detection/typescript/detect-gemini.ts
// Positive test cases for Gemini (TypeScript)
// ruleid: detect-gemini
import { GoogleGenerativeAI } from "@google/generative-ai";

// ruleid: detect-gemini
const genAI = new GoogleGenerativeAI("API_KEY");

// ruleid: detect-gemini
const model = genAI.getGenerativeModel({ model: "gemini-pro" });

// ruleid: detect-gemini
const result = await model.generateContent("Hello");

// ruleid: detect-gemini
const chat = model.startChat();

// ruleid: detect-gemini
const response = await chat.sendMessage("Hi");
